src/main/resources/templates/db_scripts/hridSettings.sql

```diff
@@ -1,17 +1,20 @@
-CREATE EXTENSION IF NOT EXISTS "pgcrypto";
-
--- ensure that there will be one row only
+-- Ensure that there will be one row only.
 ALTER TABLE ${myuniversity}_${mymodule}.${table.tableName}
   ADD COLUMN IF NOT EXISTS
     lock boolean DEFAULT true UNIQUE CHECK(lock=true);
+-- Derive id from tenant, module and table, setting UUID version to 4 and
+-- UUID variant byte to 8.
+-- Don't use pgcrypto gen_random_uuid() because pg-pool cannot replicate
+-- random numbers: https://www.pgpool.net/docs/latest/en/html/restrictions.html
 INSERT INTO ${myuniversity}_${mymodule}.${table.tableName}
   SELECT id, jsonb_build_object(
     'id', id,
     'instances', jsonb_build_object('prefix', 'in', 'startNumber', 1),
     'holdings', jsonb_build_object('prefix', 'ho', 'startNumber', 1),
     'items', jsonb_build_object('prefix', 'it', 'startNumber', 1)
   )
-  FROM (SELECT gen_random_uuid() AS id) AS alias
+  FROM (SELECT overlay(overlay(md5('${myuniversity}_${mymodule}.${table.tableName}')
+                 placing '4' from 13) placing '8' from 17)::uuid AS id) AS alias
```

Comment:
Alternatively, could we remove the `id` column entirely?

@mreno-EBSCO Why does this configuration need a UUID `id` column?

Reply:
I copied the pattern from another script, so I don't know if a UUID is required for this locking SQL to work. The pattern I copied it from generated invalid versioned UUIDs, so I changed it to the pgcrypto built in.

My problem with this is that I wasn't aware of a FOLIO requirement that module SQL must support pgpool nor can I find any indication it is a requirement in the wiki, dev site or github. All I see are vague references to pgpool here and there. I think that this requirement, if it is indeed a requirement, should be more explicit somewhere so that failure is not something that we have to wait until deployed to an environment that uses this software to discover. Then again, perhaps it is explicitly stated as a requirement somewhere that my searches have not found... 🤷‍♂️

Reply:
Removing the `id` column entirely is possible but requires several code changes:
The table `hrid_settings` is generated by a table entry in schema.json and we use some RMB methods to access it.
Therefore I suggest to simply provide some id without using gen_random_uuid().

Reply:
@mreno-EBSCO 

> My problem with this is that I wasn't aware of a FOLIO requirement that module SQL must support pgpool nor can I find any indication it is a requirement in the wiki, dev site or github.

You are not alone. As far as I'm aware, there is no official requirement for this.

> I think that this requirement, if it is indeed a requirement, should be more explicit somewhere

I totally agree

Reply:
@julianladisch 

> Removing the id column entirely is possible but requires several code changes:
The table hrid_settings is generated by a table entry in schema.json and we use some RMB methods to access it.

That is a good point. I was merely coming up with ideas. I agree this is not viable.

> Therefore I suggest to simply provide some id without using gen_random_uuid().

I agree, and I like the suggestion you made in the other comment, to use hard coded generated UUIDs.